ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -648,7 +653,7 @@ public static GetBlobOptions buildGetBlobOptions(Map<String, Object> args, SubRe
       throw new RestServiceException("Ranges not supported for sub-resources that aren't Segment.",
           RestServiceErrorCode.InvalidArgs);
     }
-    boolean resolveRangeOnEmptyBlob = getBooleanHeader(args, Headers.RESOLVE_RANGE_ON_EMPTY_BLOB, false);
+    boolean resolveRangeOnEmptyBlob = getBooleanHeader(args, Headers.RESOLVE_RANGE_ON_EMPTY_BLOB, false, false);
```

Comment:
how is this related to named blob upsert ? curious

Reply:
So I changed 'getBooleanHeader' to accept a defaultVal, and need to pass false here to be backward-compatible. Also see https://github.com/linkedin/ambry/pull/2247#discussion_r1018249962

Reply:
nvm, i see this is a side effect of the named-upsert flag